core/chains/evm/gas/models.go

```diff
@@ -164,6 +168,26 @@ func (e WrappedEvmEstimator) GetFee(ctx context.Context, calldata []byte, feeLim
 	return
 }
 
+func (e WrappedEvmEstimator) GetMaxCost(ctx context.Context, amount assets.Eth, calldata []byte, feeLimit uint32, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (*big.Int, error) {
+	fees, gasLimit, err := e.GetFee(ctx, calldata, feeLimit, maxFeePrice, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var gasPrice *assets.Wei
+	if e.EIP1559Enabled {
+		gasPrice = fees.DynamicFeeCap
+	} else {
+		gasPrice = fees.Legacy
+	}
+
+	// Creating a `Big` struct to avoid having a mutation on `amount` and hence affecting the value stored in the DB
+	amountAsBig := utils.NewBig(amount.ToInt())
+	fee := new(big.Int).Mul(gasPrice.ToInt(), big.NewInt(int64(gasLimit)))
+	amountWithFees := new(big.Int).Add(amountAsBig.ToInt(), fee)
+	return amountWithFees, nil
```

Comment:
Actually, you don't need this at all. You can simply do:

```suggestion
	fee := new(big.Int).Mul(gasPrice.ToInt(), big.NewInt(int64(gasLimit)))
	amountWithFees := new(big.Int).Add(amount.ToInt(), fee)
	return amountWithFees, nil
```